libcaf_core/caf/logger.hpp

```diff
@@ -209,53 +176,19 @@ class CAF_CORE_EXPORT logger : public ref_counted {
 
   /// Returns whether the logger is configured to accept input for given
   /// component and log level.
-  bool accepts(unsigned level, std::string_view component_name);
-
-  /// Returns the output format used for the log file.
-  const line_format& file_format() const {
-    return file_format_;
-  }
-
-  /// Returns the output format used for the console.
-  const line_format& console_format() const {
-    return console_format_;
-  }
-
-  unsigned verbosity() const noexcept {
-    return cfg_.verbosity;
-  }
-
-  unsigned file_verbosity() const noexcept {
-    return cfg_.file_verbosity;
-  }
-
-  unsigned console_verbosity() const noexcept {
-    return cfg_.console_verbosity;
-  }
+  virtual bool accepts(unsigned level, std::string_view component_name) = 0;
 
   // -- static utility functions -----------------------------------------------
 
+  /// Creates a new logger instance.
+  static logger* make(actor_system& sys);
```

Comment:
Returning a raw pointer that transfers ownership is a major red flag:

```suggestion
  /// Creates a new logger instance.
  static intrusive_ptr<caf::logger> make(actor_system& sys);
```